DNN Platform/Library/Services/Search/Internals/SearchSecurityTrimmer.cs

```diff
@@ -178,11 +178,12 @@ private void PrepareScoreDocs()
                         continue;
                     }
 
-                    _scoreDocs.Add(scoreDoc);
-                    if (++collectedSoFar == pageSize)
+                    if (++collectedSoFar >= pageSize)
```

Comment:
There is a bug here. The `collectedSoFar` counter is incremented but then the document might not be added. This fails when pageSize is 1 for example. So this should be compared using `>` instead of `>=`. But that would change the meaning of `collectedSoFar`, so to preserve the correct meaning I suggest following implementation:
```c#
                    if (collectedSoFar >= pageSize)
                    {
                        continue;
                    }

                    _scoreDocs.Add(scoreDoc);
                    collectedSoFar++;
```